File: sdks/java/src/main/java/com/boltprediction/sdk/model/CaseSimilaritySearchRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 案例相似度检索请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class CaseSimilaritySearchRequest {

    @JsonProperty("node_type")
    private Object nodeType;

    @JsonProperty("node_id")
    private Object nodeId;

    @JsonProperty("fault_type")
    private Object faultType;

    @JsonProperty("fault_level")
    private Object faultLevel;

    @JsonProperty("sensor_data")
    private Object sensorData;

    @JsonProperty("sensor_features")
    private Object sensorFeatures;

    @JsonProperty("feature_vector")
    private Object featureVector;

    @JsonProperty("tags")
    private Object tags;

    @JsonProperty("top_k")
    private Integer topK;

    @JsonProperty("min_similarity")
    private Double minSimilarity;

    @JsonProperty("only_approved")
    private Boolean onlyApproved;

    @JsonProperty("tenant_id")
    private Object tenantId;

    public Object getNodeType() {
        return nodeType;
    }

    public void setNodeType(Object nodeType) {
        this.nodeType = nodeType;
    }

    public Object getNodeId() {
        return nodeId;
    }

    public void setNodeId(Object nodeId) {
        this.nodeId = nodeId;
    }

    public Object getFaultType() {
        return faultType;
    }

    public void setFaultType(Object faultType) {
        this.faultType = faultType;
    }

    public Object getFaultLevel() {
        return faultLevel;
    }

    public void setFaultLevel(Object faultLevel) {
        this.faultLevel = faultLevel;
    }

    public Object getSensorData() {
        return sensorData;
    }

    public void setSensorData(Object sensorData) {
        this.sensorData = sensorData;
    }

    public Object getSensorFeatures() {
        return sensorFeatures;
    }

    public void setSensorFeatures(Object sensorFeatures) {
        this.sensorFeatures = sensorFeatures;
    }

    public Object getFeatureVector() {
        return featureVector;
    }

    public void setFeatureVector(Object featureVector) {
        this.featureVector = featureVector;
    }

    public Object getTags() {
        return tags;
    }

    public void setTags(Object tags) {
        this.tags = tags;
    }

    public Integer getTopK() {
        return topK;
    }

    public void setTopK(Integer topK) {
        this.topK = topK;
    }

    public Double getMinSimilarity() {
        return minSimilarity;
    }

    public void setMinSimilarity(Double minSimilarity) {
        this.minSimilarity = minSimilarity;
    }

    public Boolean getOnlyApproved() {
        return onlyApproved;
    }

    public void setOnlyApproved(Boolean onlyApproved) {
        this.onlyApproved = onlyApproved;
    }

    public Object getTenantId() {
        return tenantId;
    }

    public void setTenantId(Object tenantId) {
        this.tenantId = tenantId;
    }

}

File: sdks/python/bolt_prediction_sdk/models/case_similarity_search_request.py
"""
CaseSimilaritySearchRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class CaseSimilaritySearchRequest(SDKBaseModel):
    """案例相似度检索请求"""

    node_type: Optional[Any] = Field(description="节点类型 bolt/flange", default=None)
    node_id: Optional[Any] = Field(description="节点ID", default=None)
    fault_type: Optional[Any] = Field(description="故障类型", default=None)
    fault_level: Optional[Any] = Field(description="故障级别", default=None)
    sensor_data: Optional[Any] = Field(description="传感器时序数据", default=None)
    sensor_features: Optional[Any] = Field(description="传感器特征", default=None)
    feature_vector: Optional[Any] = Field(description="特征向量", default=None)
    tags: Optional[Any] = Field(description="标签过滤", default=None)
    top_k: Optional[int] = Field(description="返回Top-K相似案例", default=5)
    min_similarity: Optional[float] = Field(description="最低相似度阈值", default=0.0)
    only_approved: Optional[bool] = Field(description="只返回已审核通过的案例", default=True)
    tenant_id: Optional[Any] = Field(description="租户ID过滤", default=None)
